app/models/job_proxy_dispatcher.rb

```diff
@@ -197,10 +230,13 @@ def busy_proxies
     end
   end
 
-  def active_scans
+  def active_scans(target_class = VmOrTemplate)
+    class_name = target_class.base_class.name
     @active_scans ||= begin
       actives = Hash.new(0)
-      actives[@zone] = Job.where(:zone => @zone, :dispatch_status => "active").where.not(:state => "finished").count
+      actives[@zone] = Job.where(:zone => @zone, :dispatch_status => "active")
```

Comment:
We're going to have collision here due to the `||=`.  We need `@active_scans` to be an object that stores a memoized list of active scans per target class.  Maybe start with a hash or create an object that does something like this:

```ruby
@active_scans ||= {}
class_name = target_class.base_class.name
@active_scans[class_name] ||= begin
  ...
```